What's New in Accessibility in the .NET Framework

rpetrusha opened the pull request:
# What's New in Accessibility in the .NET Framework 

## Summary

Since accessibility is a major focus, it makes sense to add a separate what's new topic for it.

The topic is published on our review site at https://review.docs.microsoft.com/en-us/dotnet/framework/whats-new/whats-new-in-accessibility?branch=pr-en-us-3522.

## Suggested Reviewers
@preetikr, @vivmishra, @Tanya-Solyanik  , @merriemcgaw

Commit:
Added topic

Commit:
Initial checkin

Commit:
Initial checkin

Commit:
Interim checkin

Commit:
Restored original what's new

Commit:
Merge branch 'master' into accessibility

Merging origin/master into accessibility.

Commit:
Merge remote-tracking branch 'upstream/master' into accessibility

Merge conflicts.

Commit:
Additional content added

Commit:
First draft completed

Tanya-Solyanik commented on docs/framework/whats-new/whats-new-in-accessibility.md:
```diff
@@ -0,0 +1,255 @@
+
+**Screen reader improvements**
+
+If accessibility improvcements are enabled, the .NET Framework 4.7.1 includes the following enhancements that affect screen readers:
```
typo

Tanya-Solyanik commented on docs/framework/whats-new/whats-new-in-accessibility.md:
````diff
@@ -0,0 +1,255 @@
+</runtime>
+```
+
+Similarly, applications that target versions of the .NET Framework starting with 4.7.1 can disable accessibility features adding the following switch to the [`<AppContextSwitchOverrides>`](~/docs/configure-apps/file-schema/runtime/appcontextswitchoverrides-element.md) element in the [`<runtime>`](~/docs/configure-apps/file-schema/runtime/index.md) section of the application's configuration file. 
````
should it be "by adding"?

Tanya-Solyanik commented on docs/framework/whats-new/whats-new-in-accessibility.md:
```diff
@@ -0,0 +1,255 @@
+    ![Disabled text after accessibility improvements](media/wf-disabled-after.png) 
+- High contrast improvements in the Thread Exception Dialog.
+
+- Better multi-monitor DPI awareness in control anchoring and scaling.
```
This item is not a part of accessibility work

Tanya-Solyanik commented on docs/framework/whats-new/whats-new-in-accessibility.md:
```diff
@@ -0,0 +1,255 @@
+
+**Enhanced UI accessibility patterns**
+
+Starting with the .NET Framework 4.7.1, developers of accessibility technology tools can leverage common api accessibility patterns and properties for several WinForms controls. These accessibility improvements include:
```
I thought that API is usually spelled in all-caps, no?

Tanya-Solyanik replied:
it should be "Enhanced support for UIAutomation accessibility patterns" - this is how WPF calls UIA in the same doc

merriemcgaw replied:
Agree with @Tanya-Solyanik - "Enhanced support for UIAutomation accessibility patterns" sounds much better.

merriemcgaw commented on docs/framework/whats-new/whats-new-in-accessibility.md:
```diff
@@ -0,0 +1,255 @@
+# What's new in accessibility in the .NET Framework
+
+The .NET Framework aims at making applications more accessibile for your users. Accessibility features allow an application to handle a range of devices and form factors seamlessly. Starting with the .NET Framework 4.7.1, the .NET Framework includes a large number of accessibility improvements that allow developers to create accessible applications. 
+
```
Could you say something about supporting Assistive Technology apps here?  It's not so much about form factors as about providing the right experience for users of assistive technology.

merriemcgaw commented on docs/framework/whats-new/whats-new-in-accessibility.md:
```diff
@@ -0,0 +1,255 @@
+
+The .NET Framework aims at making applications more accessibile for your users. Accessibility features allow an application to handle a range of devices and form factors seamlessly. Starting with the .NET Framework 4.7.1, the .NET Framework includes a large number of accessibility improvements that allow developers to create accessible applications. 
+
+The new accessibility features are enabled by default for applications that target the .NET Framework 4.7.1 or later. In addition, applications that target an earlier version of the .NET Framework but are running on the .NET Framework 4.7.1 or later can opt out of legacy accessibility behaviors by adding the following switch to the [`<AppContextSwitchOverrides>`](~/docs/configure-apps/file-schema ...
```
This isn't super clear that opting out of legacy behaviors is the same thing as opting in to the 4.7.1 improvements.

merriemcgaw commented on docs/framework/whats-new/whats-new-in-accessibility.md:
```diff
@@ -0,0 +1,255 @@
+    After:
+
+    ![Disabled text after accessibility improvements](media/wf-disabled-after.png) 
+- High contrast improvements in the Thread Exception Dialog.
```
Would a before/after image be helpful here?

rpetrusha replied:
There is a before/after image here. Sorry, @merriemcgaw  -- I should have included a link to the staged published topic. It is [What's new in accessibility in the .NET Framework](https://review.docs.microsoft.com/en-us/dotnet/framework/whats-new/whats-new-in-accessibility?branch=pr-en-us-3522).

Commit:
Addressed review comments, fixed links

rpetrusha commented:
@Tanya-Solyanik and @merriemcgaw, thanks for your review. I've addressed all of your comments. When the build finishes, the staged topic is available [here](https://review.docs.microsoft.com/en-us/dotnet/framework/whats-new/whats-new-in-accessibility?branch=pr-en-us-3522).

BillWagner commented:
@rpetrusha Can you fix the merge conflicts here? Then, I'll give it a review.

Commit:
Merge branch 'master' into accessibility

Commit:
Merge branch 'master' into accessibility

rpetrusha commented:
Thanks, @BillWagner. I've resolved the merge conflict.

JRAlexander commented:
I think you need to make this MVC compliant.

BillWagner commented on docs/framework/whats-new/whats-new-in-accessibility.md:
````diff
@@ -0,0 +1,254 @@
+```
+
+When the data in the live region changes and you need to inform a screen reader, you explicitly raise an event, as shown in the following sample.
+
````
Should these be in tabs?

BillWagner commented on docs/framework/whats-new/whats-new-in-accessibility.md:
```diff
@@ -0,0 +1,254 @@
+
+**High contrast**
+
+Starting with the .NET Framework 4.7.1, improvements in high conrast have been made to various WPF controls. They are now visible when the <xref:System.Windows.SystemParameters.HighContrast%2A> theme is set. These include:
```
typo: conrast => contrast.

rpetrusha replied:
Thanks, @BillWagner. I'll add it to the TOC.

BillWagner reviewed (changes requested):
Overall, this looks great.

I found 1 typo, and had one suggestion.

Also, this new article isn't in the TOC. Do you want to add it, or only have it available from the main "What's new" topic?

Commit:
Addressed review comments

Commit:
Merge branch 'master' into accessibility

Commit:
Merge branch 'master' into accessibility

rpetrusha commented:
I've addressed your comments, @BillWagner. The accessibility topic already was in the TOC as a child of the What's New in the .NET Framework topic. Should it be a peer rather than a child?

mairaw commented:
I think it's fine as a child like it's showing on review now.

BillWagner commented:
@rpetrusha It's fine. I reviewed the diffs, and I thought it was a new file, but I didn't see a TOC change. I say :shipit: when ready.
